frontend/src/components/DefenceBox/DefenceMechanism.tsx

```diff
@@ -69,73 +69,77 @@ function DefenceMechanism({
 		}
 	}
 	return (
-		<details
-			className="defence-mechanism"
-			onToggle={() => {
-				// re-render the configuration component when detail is toggled
-				// this is to resize the textarea when detail is expanded
-				setConfigKey(configKey + 1);
-			}}
-		>
-			<summary>
-				<span aria-hidden={defenceDetail.id !== DEFENCE_ID.PROMPT_ENCLOSURE}>
-					{defenceDetail.name}
-				</span>
-				{defenceDetail.id !== DEFENCE_ID.PROMPT_ENCLOSURE && (
-					<label className="switch">
+		<fieldset className="defence-mechanism-fieldset">
+			<legend className="defence-mechanism-legend">{defenceDetail.name}</legend>
+			{defenceDetail.id !== DEFENCE_ID.PROMPT_ENCLOSURE && (
+				<form className="defence-mechanism-form">
```

Comment:
I'm curious about the rationale of wrapping the toggles `<div>` within a form.

Reply:
feel free to have a play around with it and see if you can find an alternative, but this part was taken from the example code, link is in the css

Reply:
The alternative is to remove the `<form>` element and put the styles onto the `<div>` instead.

![image](https://github.com/ScottLogic/prompt-injection/assets/118171430/cda59ce5-810e-4e4f-9111-5224340c1439)

![image](https://github.com/ScottLogic/prompt-injection/assets/118171430/ac7322dc-a34a-46f1-8c0d-9f493b6fcf7b)

looks and works the same way.

As I understand `<form>`s are for (like their namesake) grouping user input elements together that will all be submitted and processed at once. So I expect the example in [the link](https://adrianroselli.com/2019/03/under-engineered-toggles.html) grouped the toggles in a form because that's how they expected it to be used in real life, in a form setting. Given we're not batch-input-and-processing any information, we may as well remove the form, since it is just wrapping a single element.

On that note, in the [updated intro](https://adrianroselli.com/2019/03/under-engineered-toggles.html) of the webpage it mentions the cases where it's better to use a button for the toggle instead of a checkbox, and it seems as though our app meets all three criteria for using a button, and none (possibly one - I don't know what progressively enhancing the control means) of those for using a checkbox. What do you think?

![image](https://github.com/ScottLogic/prompt-injection/assets/118171430/baefb755-ce04-468e-88a2-87599cfd1818)